gcache/src/gcache_params.cpp

```diff
@@ -57,21 +57,15 @@ gcache::GCache::Params::register_params(gu::Config& cfg)
 #ifndef NDEBUG
     cfg.add(GCACHE_PARAMS_DEBUG,           GCACHE_DEFAULT_DEBUG);
 #endif
-<<<<<<< HEAD
-    cfg.add(GCACHE_PARAMS_RECOVER,         GCACHE_DEFAULT_RECOVER);
+    cfg.add(GCACHE_PARAMS_RECOVER, GCACHE_DEFAULT_RECOVER,
+            gu::Config::Flag::read_only | gu::Config::Flag::type_bool);
 #ifdef PXC
     cfg.add(GCACHE_PARAMS_KEEP_PAGES_COUNT, GCACHE_DEFAULT_KEEP_PAGES_COUNT);
     cfg.add(GCACHE_PARAMS_FREEZE_PURGE_SEQNO, GCACHE_DEFAULT_FREEZE_PURGE_SEQNO);
 #endif /* PXC */
     cfg.add(GCACHE_PARAMS_ENCRYPTION, GCACHE_DEFAULT_ENCRYPTION);
     cfg.add(GCACHE_PARAMS_ENCRYPTION_CACHE_PAGE_SIZE, GCACHE_DEFAULT_ENCRYPTION_CACHE_PAGE_SIZE);
     cfg.add(GCACHE_PARAMS_ENCRYPTION_CACHE_SIZE, GCACHE_DEFAULT_ENCRYPTION_CACHE_SIZE);
```

Comment:
Please check

- `GCACHE_PARAMS_ENCRYPTION` , `GCACHE_PARAMS_ENCRYPTION_CACHE_PAGE_SIZE` and `GCACHE_PARAMS_ENCRYPTION_CACHE_SIZE` must be inside `#ifdef PXC` section as these exist only in PXC only.
- Add `gu::Config::Flag::type_bool` flag to `GCACHE_PARAMS_ENCRYPTION`
- Add `gu::Config::Flag::type_integer` flag to `GCACHE_PARAMS_ENCRYPTION_CACHE_PAGE_SIZE` and `GCACHE_PARAMS_ENCRYPTION_CACHE_SIZE` and set the read only flags if necessary.

Reply:
Also please set the appropriate flags for `GCACHE_PARAMS_KEEP_PAGES_COUNT` and `GCACHE_PARAMS_FREEZE_PURGE_SEQNO` as well.

Reply:
Also, the same needs to be done in Allocator params as well. See `gu::Allocator::register_params()` in `galerautils/src/gu_alloc.cpp`.

Reply:
Done